plugins/device/rest.go

```diff
@@ -104,3 +104,28 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, peerInfo)
 }
+
+func deviceListHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Invalid peer-id passed in url")
+		return
+	}
+
+	devices, err := GetDevices(peerID)
```

Comment:
Unrelated to this PR: I had a look at `GetDevices` which does `return nil, nil` at the end. That can be improved as follows:

```go
func GetDevices(peerID string) ([]deviceapi.Info, error) {

        peerInfo, err := peer.GetPeer(peerID)
        if err != nil {
                return nil, err
        }

        var deviceInfo []deviceapi.Info
        if devices, ok := peerInfo.MetaData["devices"]; ok {
                if err := json.Unmarshal([]byte(devices), &deviceInfo); err != nil {
                        return nil, err
                }
        }

        return deviceInfo, nil
}
```

As you have to update the PR to address the comment below, you might as well fix this one as well.

Reply:
@rishubhjain The code on `GetDevices` uses `devices` as key, it's `_devices` at other places. Please find places where it's `devices` and change them to `_devices`